xsc/services/version.go

```diff
@@ -0,0 +1,51 @@
+package services
+
+import (
+	"encoding/json"
+	"net/http"
+	"strings"
+
+	"github.com/jfrog/jfrog-client-go/auth"
+	"github.com/jfrog/jfrog-client-go/http/jfroghttpclient"
+	"github.com/jfrog/jfrog-client-go/utils"
+	"github.com/jfrog/jfrog-client-go/utils/errorutils"
+)
+
+// VersionService returns the https client and Xsc details
+type VersionService struct {
+	client     *jfroghttpclient.JfrogHttpClient
+	XscDetails auth.ServiceDetails
+}
+
+// NewVersionService creates a new service to retrieve the version of Xsc
+func NewVersionService(client *jfroghttpclient.JfrogHttpClient) *VersionService {
+	return &VersionService{client: client}
+}
+
+// GetXscDetails returns the Xsc details
+func (vs *VersionService) GetXscDetails() auth.ServiceDetails {
+	return vs.XscDetails
+}
+
+// GetVersion returns the version of Xsc
+func (vs *VersionService) GetVersion() (string, error) {
+	httpDetails := vs.XscDetails.CreateHttpClientDetails()
+	resp, body, _, err := vs.client.SendGet(vs.XscDetails.GetUrl()+"api/v1/system/version", true, &httpDetails)
```

Comment:
I encountered an error of constructing the wrong url. If we get the url without '/' at the end the url will be constructed wrong. maybe it is better to add some check wherever we construct a url: if the url doesnt end with '/', add it before we concatenate it with the suffix

Reply:
There is an existing function for this: AddTrailingSlashIfNeeded

Reply:
how did you provide the url? cause in the flags+"jf c add" mechanism we have functionality that is supposed to protect us from getting bad url suffixes.
we use those GET/POST/PUT http requests all over the code without taking care of the slashes.